extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/variant-analysis.integration.test.ts

```diff
@@ -0,0 +1,142 @@
+import * as path from 'path';
+
+import * as sinon from 'sinon';
+
+import { commands, extensions, TextDocument, window, workspace } from 'vscode';
+import { setupServer } from 'msw/node';
+import * as Octokit from '@octokit/rest';
+import { retry } from '@octokit/plugin-retry';
+
+import { CodeQLExtensionInterface } from '../../../extension';
+import { createRequestHandlers } from '../../../mocks/request-handlers';
+import * as config from '../../../config';
+import { Credentials } from '../../../authentication';
+
+const server = setupServer();
+
+before(() => server.listen());
+
+afterEach(() => server.resetHandlers());
+
+after(() => server.close());
+
+async function loadScenario(scenarioName: string) {
+  const handlers = await createRequestHandlers(path.join(__dirname, '../../../../src/mocks/scenarios', scenarioName));
+
+  server.use(...handlers);
+}
+
+async function showQlDocument(name: string): Promise<TextDocument> {
+  const folderPath = workspace.workspaceFolders![0].uri.fsPath;
+  const documentPath = path.resolve(folderPath, name);
+  const document = await workspace.openTextDocument(documentPath);
+  await window.showTextDocument(document!);
+  return document;
+}
+
+describe('Variant Analysis Integration', function() {
```

Comment:
I guess these are tests around variant analysis submission. What do you think?

Reply:
Yes, that makes sense. The reason for calling it an integration test is that they are not linked to a specific class, so if I had just named the file `variant-analysis.test.ts` I'd expect it to test a file `variant-analysis.ts`. To differentiate, I added `.integration.test.ts` to ensure that the difference is clear. What do you think?

Reply:
I think once we start talking about integration of things it makes sense to not have a 1-to-1 mapping between tests and classes. We're testing the integration between multiple things to prove out a feature or part of it. In this case we're testing variant analysis submission. 

So perhaps `variant-analysis-submission-integration.test.ts`?